flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/RoundRobinOperatorStateRepartitioner.java

```diff
@@ -123,9 +128,28 @@
         return mergeMapList;
     }
 
-    /** Collect union states from given parallelSubtaskStates. */
     private Map<String, List<Tuple2<StreamStateHandle, OperatorStateHandle.StateMetaInfo>>>
             collectUnionStates(List<List<OperatorStateHandle>> parallelSubtaskStates) {
+        return collectStates(parallelSubtaskStates, OperatorStateHandle.Mode.UNION);
+    }
+
+    private Map<String, List<Tuple2<StreamStateHandle, OperatorStateHandle.StateMetaInfo>>>
+            collectPartlyFinishedBroadcastStates(
+                    List<List<OperatorStateHandle>> parallelSubtaskStates) {
+        return collectStates(parallelSubtaskStates, OperatorStateHandle.Mode.BROADCAST).entrySet()
+                .stream()
+                .filter(
+                        e ->
+                                e.getValue().size() > 0
+                                        && e.getValue().size() < parallelSubtaskStates.size())
```

Comment:
`e.getValue().size() > 0` means `not empty`?
what does `e.getValue().size() < parallelSubtaskStates.size()` mean?

Reply:
My guess is that the empty state should be `e.getValue().size() == 0` ?

Reply:
Here is to check if all the subtasks have reported the state, so we could filter out the partially finished ones. But the condition here indeed has some problems and I changed it.